dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+  final double width = parsePixels(_extractAttr(svgElement, 'width')).toDouble();
+  final double height = parsePixels(_extractAttr(svgElement, 'height')).toDouble();
+
+  final List<SvgPath> paths =
+    _interpretSvgGroup(svgElement.children, new _Transform());
+  return new FrameData(new Point<double>(width, height), paths);
+}
+
+List<SvgPath> _interpretSvgGroup(List<XmlNode> children, _Transform transform) {
+  final List<SvgPath> paths = <SvgPath>[];
+  for (XmlNode node in children) {
+    if (node.nodeType != XmlNodeType.ELEMENT)
+      continue;
+    final XmlElement element = node;
+
+    if (element.name.local == 'path') {
+      // TODO(amirh): convert relative commands to absolute
+      paths.add(SvgPath.fromElement(element).applyTransform(transform));
+    }
+
+    if (element.name.local == 'g') {
+      double opacity = transform.opacity;
+      if (_hasAttr(element, 'opacity'))
+        opacity *= double.parse(_extractAttr(element, 'opacity'));
+
+      Matrix3 transformMatrix = transform.transformMatrix;
+      if (_hasAttr(element, 'transform'))
+        transformMatrix = transformMatrix.multiplied(
+            _parseSvgTransform(_extractAttr(element, 'transform')));
+
+      final _Transform subtreeTransform = new _Transform(
+        transformMatrix: transformMatrix,
+        opacity: opacity
+      );
+      paths.addAll(_interpretSvgGroup(element.children, subtreeTransform));
+    }
+  }
+  return paths;
+}
+
+// Given a points list in the form e.g: "25.0, 1.0 12.0, 12.0 23.0, 9.0" matches
+// the coordinated of the first point and the rest of the string, for the
+// example above:
+// group 1 will match "25.0"
+// group 2 will match "1.0"
+// group 3 will match "12.0, 12.0 23.0, 9.0"
+final RegExp _pointMatcher = new RegExp(r' *([\-\.0-9]+) *, *([\-\.0-9]+)(.*)');
+
+/// Parse a string with a list of points, e.g:
+/// '25.0, 1.0 12.0, 12.0 23.0, 9.0' will be parsed to:
+/// [Point(25.0, 1.0), Point(12.0, 12.0), Point(23.0, 9.0)].
+List<Point<double>> parsePoints(String points) {
+  String unParsed = points;
+  final List<Point<double>> result = <Point<double>>[];
+  while(unParsed.isNotEmpty && _pointMatcher.hasMatch(unParsed)) {
+    final Match m = _pointMatcher.firstMatch(unParsed);
+    result.add(new Point<double>(
+        double.parse(m.group(1)),
+        double.parse(m.group(2))
+    ));
+    unParsed = m.group(3);
+  }
+  return result;
+}
+
+/// Data for a single animation frame.
+class FrameData {
+  const FrameData(this.size, this.paths);
+
+  final Point<double> size;
+  final List<SvgPath> paths;
+
+  @override
+  bool operator ==(Object other){
+    if (runtimeType != other.runtimeType)
+      return false;
+    final FrameData typedOther = other;
+    return size == typedOther.size
+        && const ListEquality<SvgPath>().equals(paths, typedOther.paths);
+  }
+
+  @override
+  int get hashCode =>
+      size.hashCode ^
+      paths.hashCode;
+
```

Comment:
as far as i can tell, this class doesn't need operator == and hashCode. I'd drop them.

Reply:
oh, i see, you use this in the tests